### PR TITLE
Remove pointless use of atomics in error macros

### PR DESCRIPTION
**Pull request by RedworkDE**
- The existing use of atomics did not prevent multiple threads from entering the condition and printing the deprecation message, as the flag was only set after the first thread finished.
- `WARN_DEPRECATED` is unused and `WARN_DEPRECATED_MSG` is only used once, in a place that most likely isn't thread safe either: https://github.com/godotengine/godot/blob/830b54a03b480f29e60b4ebc813c1e24ac140a70/scene/resources/mesh.cpp#L901-L903
- The `*_ONCE` macros that are actually used all over the place have no such threading protection.
- The worst failure that can happen, is that the error gets printed a handful of times (once by every thread that reaches it at the same time)
- If the errors should really never under any circumstances be printed twice, I will rework this PR to correctly ensure this using atomics and exchange.
- In the mean time I reordered setting the flag to come before printing the warning, which should make it pretty much impossible to trigger duplicated messages, even in constructed scenarios
- Made the `WARN_DEPRECATED*` and `*_ONCE` macros use the same guard pattern as there was no real reason for the difference
- (Maybe) avoid including atomic in some places (SafeRefCounted includes it pretty much everywhere anyways)
- Also fix one typo while I was editing this file
- In summary: the pointless / incorrect use of atomics annoyed me, so I changed it

### CODE REVIEW

**Review by Ivorforce — APPROVED** [review]
This seems sensible to me. It's nice to remove the `<atomic>` include from a file that's included across half the codebase (although notably, it's still understandably included a lot through `safe_refcount`).

I agree the use of atomics was faulty before. I'm not sure if the rename of `first_print` is needed, but it _is_ slightly easier to understand after.
Moving the `warning_shown` to before the print makes sense too, though it's not super likely to have an impact. As said in the OP, worst case it will print one or two more times than needed, on the exceedingly rare incident where two threads access the same print at the same time. I don't think this is worth the overhead of atomics.

Note: Since @RedworkDE has been inactive for Godot for a while, I took the liberty to rebase (there was one trivial collision for the comment at the atomics include).

**Review by akien-mga — APPROVED** [review]
Makes sense to me too.

**Comment by akien-mga** [comment]
Thanks!
